libbuteosyncfw/common/Logger.cpp

```diff
@@ -21,215 +22,38 @@
  *
  */
 
+#include <QScopedPointer>
 
 #include "Logger.h"
-#include <QMutexLocker>
-#include <iostream>
-#include <QFile>
-#include <QFileInfo>
-#include <QDir>
 
-#include <syslog.h>
+#define TRACE_FUNCTION_TIME 0
 
 using namespace Buteo;
 
-const int Logger::DEFAULT_INDENT_SIZE = 4;
-
-Logger *Logger::sInstance = nullptr;
-
-Logger *Logger::instance()
-{
-    if (nullptr == sInstance) {
-        createInstance();
-    }
-    return sInstance;
-}
-
-int Logger::defaultLogLevel()
-{
-    QByteArray levelByteArray = qgetenv("MSYNCD_LOGGING_LEVEL");
```

Comment:
Could be an option to consider should we still support this for enabling debug.

Reply:
Is that support existing scripts? I'm a bit reluctant to support the old usage if the Qt logging categories can be used instead, but if there's a good reason to, we could check for this env var when running msyncd or the out-of-process plugin runner.

Reply:
Not necessarily even scripts, but we've been documenting this environment variable on different places and it getting the change to everyone might take some time. Compare to EAS which uses qt logging categories but has kept the old configuration for setting up the category levels.

I'm ok either way, just something to consider.

Reply:
Yes that's true, would be useful to retain some support for it. Added this commit: https://github.com/sailfishos/buteo-syncfw/pull/3/commits/fc4661eea14cbbb3b9537a6f952be316d962edfb